rest/polygon.go

```diff
@@ -15,16 +15,18 @@ type polygonClient struct {
 	Reference *reference.Client
 	Trades    *trades.Client
 	Snapshot  *snapshot.Client
+	Client    *client.Client
```

Comment:
you could "inherit" here (`client.Client` instead of `Client *client.Client`). that would make using it a bit cleaner (`c.NewRequest(...)` instead of `c.Client.NewRequest(...)`)